packages/snaps-sdk/src/jsx/components/Icon.ts

```diff
@@ -0,0 +1,43 @@
+import { createSnapComponent } from '../component';
+
+/**
+ * The props of the {@link Icon} component.
+ *
+ * @property name - The name of the icon to display from a pre-defined list.
+ * @property color - The color of the displayed icon.
+ * @property size - The size of the displayed icon. Use `inherit` to size it the same as the text.
+ */
+export type IconProps = {
+  name: string;
```

Comment:
Hmm, should we add the known icons at the time of commit at least to the type? 🤔 

Any plan for what to show if the icon name is invalid?

cc @Mrtenz

Reply:
There's not 😅

But we could just copy the names from: https://github.com/MetaMask/metamask-extension/blob/develop/ui/components/component-library/icon/icon.types.ts#L28

It's definitely not ideal, but gives a bit better dev-ex